File: 0086-partition-list/0086-partition-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* partition(struct ListNode* head, int x) {
    if(head==NULL || head->next==NULL){
        return head;
    }
   
    struct ListNode*left=(struct ListNode*)malloc(sizeof(struct ListNode));
    struct ListNode*right=(struct ListNode*)malloc(sizeof(struct ListNode));
    left->next=NULL;
    right->next=NULL;
    
    struct ListNode*Lleft=left;
    struct ListNode*Rright=right;
    
    while(head!=NULL){
        if(head->val < x){
            Lleft->next=head;
            Lleft=Lleft->next;
        }
        else{
            Rright->next=head;
            Rright=Rright->next;
        }
        head=head->next;
    }
    
    Lleft->next=right->next;
    Rright->next=NULL;
    
    return left->next;
}

File: 0086-partition-list/0086-partition-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode partition(ListNode head, int x) {
        ListNode small=new ListNode(0);
        ListNode high=new ListNode(0);
        
        ListNode smallptr=small;
        ListNode highptr=high;
        
        while(head!=null){
            if(head.val < x){
                smallptr.next=head;
                smallptr=smallptr.next;
            }
            else{
                highptr.next=head;
                highptr=highptr.next;
            }
            head=head.next;
        }
        smallptr.next=high.next;
        highptr.next=null;
        
        return small.next;
    }
}